test/types/community/slow-model.test-d.ts

```diff
@@ -0,0 +1,412 @@
+  unit____ = 'unit____',
+  unit_____ = 'unit_____'
+}
+
+export interface ExtraType10 {
+  prop34?: UnitType3;
+  prop35?: number;
+  prop36?: number;
+  prop37?: boolean;
+}
+
+export interface ExtraType11 {
+  prop38?: number;
+  prop39?: boolean;
+  prop40?: boolean;
+  prop41?: boolean;
+}
+
+export interface ExtraType12 {
+  _id?: UUID;
+  prop42: ExtraType13[];
+}
+
+export interface ExtraType14 {
+  prop42: TimeType;
+  prop43?: TimeType | null;
+  prop44?: LengthType2;
+}
+
+export type ExtraType15 = string | number | boolean | ExtraType14 | TimeType | LengthType2;
+export type ExtraType16 = {
+  prop45: string | number;
+  prop46: ExtraType15;
+  prop47: ExtraType15;
+  prop48: string;
+};
+export type ExtraType13 = ExtraType15 | ExtraType16;
+
+export interface ExtraType6 {
+  prop49: SpecialType[];
+  prop50: string;
+}
+
+export interface SlowModel extends SlowModelBase {
+  _id: UUID;
+}
+
+export interface SpecialType extends SlowModel {
+  prop51: string;
+}
+
+export enum ExtraType17 {
+  Key16 = 'key16',
+  Key17 = 'key17'
+}
+
+export interface IType1 {
+  type: ExtraType17;
+}
+export interface Key17IType1 extends IType1 {
+  type: ExtraType17.Key17;
+  prop52: string;
+  prop53?: string;
+  prop54?: string;
+}
+export interface Key16IType1 extends IType1 {
+  type: ExtraType17.Key16;
+  prop55: string;
+  prop56: string | null;
+}
+
+export type ExtraType4 = Key17IType1 | Key16IType1;
+
+export enum ExtraType18 {
+  Key18 = 'key18',
+  Key19 = 'key19'
+}
+
+interface IType2 {
+  type: ExtraType18;
+}
+export type ExtraType5 = Key18IType2 | Key19IType2;
+
+export interface Key18IType2 extends IType2 {
+  type: ExtraType18.Key18;
+  prop57: string;
+  prop58: string;
+  prop59: boolean;
+  prop60: string | null;
+  prop61: boolean;
+  prop62?: string;
+}
+export interface Key19IType2 extends IType2 {
+  type: ExtraType18.Key19;
+  prop63: string;
+  prop64?: string;
+  prop65: boolean;
+}
+
+export type ExtraType19 = ExtraType15[] | ExtraType20;
+export interface ExtraType7<T extends ExtraType19 = ExtraType19> {
+  id: UUID;
+  prop66?: string;
+  prop67: T;
+  prop68?: ExtraType21;
+  prop69?: ExtraType22;
+  prop70?: string;
+}
+
+export interface ExtraType20 {
+  prop71: string;
+  prop72: string;
+  prop73: number;
+  prop74: SpecialType[];
+}
+
+export interface ExtraType22 {
+  prop75: string;
+}
+
+export interface ExtraType21 {
+  prop76: string;
+  prop77: string;
+  prop78: boolean;
+  prop79: boolean;
+  prop80: string | null;
+}
+
+export interface ExtraType8 {
+  id: UUID;
+  prop81: string;
+  prop82: ExtraType15[];
+  prop83: ExtraType15;
+}
+
+export class ExampleImpl {
+  private circularCol: Collection<CircularModel>;
+  private slowCol: Collection<SlowModel>;
+
+  constructor(db: Db) {
+    this.circularCol = db.collection('circular');
+    this.slowCol = db.collection('slow');
+  }
+
+  public method = async (): Promise<void> => {
+    /*
+     * This call shows the base issue with recursive types
```

Comment:
we should add a comment at the top crediting the user with the test code

Reply:
Good call! Added